composer/core/data_spec.py

```diff
@@ -174,14 +175,30 @@ def __init__(
             else:
                 self.num_samples = None
 
-        if isinstance(dataloader, torch.utils.data.DataLoader) and dataloader._iterator is not None:
-            raise ValueError(
-                ('The dataloader has an active iterator. This could occur '
-                 'if `persistent_workers=True` and the dataloader has already been iterated, '
-                 'or if the dataloader is mid-epoch. It is required that the training dataloader '
-                 'does not have an active iterator, so CPU dataset augmentations can be '
-                 'correctly inserted. To fix, please do not iterate over the dataloader before passing it into '
-                 'the Trainer.'))
+        if isinstance(dataloader, torch.utils.data.DataLoader):
+            if dataloader._iterator is not None:
+                raise ValueError(
+                    ('The dataloader has an active iterator. This could occur '
+                     'if `persistent_workers=True` and the dataloader has already been iterated, '
+                     'or if the dataloader is mid-epoch. It is required that the training dataloader '
+                     'does not have an active iterator, so CPU dataset augmentations can be '
+                     'correctly inserted. To fix, please do not iterate over the dataloader before passing it into '
+                     'the Trainer.'))
+            world_size = dist.get_world_size()
+            if world_size > 1 and not (
+                (dataloader.sampler and isinstance(dataloader.sampler, DistributedSampler)) or
```

Comment:
So I just realized that (maybe?) with our streaming dataloader, a `DistributedSampler` should not be set: https://github.com/mosaicml/composer/blob/dev/composer/datasets/imagenet_hparams.py#L190

Reply:
Per offline discussion, looks like we need a check to make sure its not an IterableDataset.